test/TrieTest.py

```diff
@@ -11,9 +11,18 @@ def setUp(self):
 
     def test_basic_retrieval(self):
         trie = Trie()
-        trie.insert("restaurant")
-        results = list(trie.lookup("restaurant"))
+        trie.insert("restaurant", "Concept")
+        trie.insert("rest", "Concept")
+        trie.insert("restaurant2", "Fast")
+        results = list(trie.gather("restaurant"))
         assert len(results) == 1
+        assert trie.root.entities() == ['root', 'Concept', 'Fast']
```

Comment:
What I meant was "write a separate test for missing entities", as opposed to conflating it with the basic retrieval test, or by replacing the basic retrieval test.